File: Samples/ApplicationResources/cs/AppResourceClassLibrary/class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Windows.ApplicationModel.Resources;
namespace AppResourceClassLibrary
{
    public static class LocalizedNamesLibrary
    {
        static ResourceLoader  resourceLoader = null;

        public static string LibraryName { 
            get 
            {
                String name;
                GetLibraryName("string1", out name);
                return name;
            } 
        }

        private static void GetLibraryName(string resourceName, out string resourceValue)
        {
            if (resourceLoader == null)
            {
                resourceLoader = ResourceLoader.GetForCurrentView("AppResourceClassLibrary/Resources");
            }
            resourceValue = resourceLoader.GetString(resourceName);
        }

    }
}

File: Samples/ApplicationResources/vb/AppResourceClassLibrary/class1.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Windows.ApplicationModel.Resources

Namespace Global.AppResourceClassLibrary

    Public Module LocalizedNamesLibrary

        Dim resourceLoader As ResourceLoader = Nothing

        Public ReadOnly Property LibraryName As String
            Get
                Dim name As String = Nothing
                GetLibraryName("string1", name)
                Return name
            End Get
        End Property

        Private Sub GetLibraryName(resourceName As String, ByRef resourceValue As String)
            If resourceLoader Is Nothing Then
                resourceLoader = ResourceLoader.GetForCurrentView("AppResourceClassLibrary/Resources")
            End If

            resourceValue = resourceLoader.GetString(resourceName)
        End Sub
    End Module
End Namespace
